src/game/collision-mediator.ts

```diff
@@ -0,0 +1,155 @@
+import { eventDestroyProjectile } from "../events/game-event";
+import { Vector } from "../utility/vector";
+import { GameContext } from "./game-context";
+import { Obstacle } from "./obstacle";
+import { Player } from "./player";
+import { Projectile } from "./projectile";
+
+export class CollisionMediator {
```

Comment:
I don't like the mediator being static, in the future, we'll probably want to instantiate it with the event emitter and other instance-bound properties.

Reply:
If it gains state in the future, sure. Right now it has no internal state.